diff -Naur -x '*~' libgphoto2-2.5.0/libgphoto2_port/usbdiskdirect/Makefile-files libgphoto2-2.5.0-fix-linking/libgphoto2_port/usbdiskdirect/Makefile-files
--- libgphoto2-2.5.0/libgphoto2_port/usbdiskdirect/Makefile-files	2010-05-28 20:51:59.000000000 +0200
+++ libgphoto2-2.5.0-fix-linking/libgphoto2_port/usbdiskdirect/Makefile-files	2012-08-04 20:55:17.066546511 +0200
@@ -4,5 +4,5 @@
 usbdiskdirect_la_LDFLAGS = $(iolib_ldflags)
 usbdiskdirect_la_CPPFLAGS = $(AM_CPPFLAGS) $(INTL_CFLAGS) $(CPPFLAGS)
 usbdiskdirect_la_DEPENDENCIES = $(iolib_dependencies)
-usbdiskdirect_la_LIBADD =  $(iolib_libadd) $(INTLLIBS)
+usbdiskdirect_la_LIBADD =  $(iolib_libadd) $(INTLLIBS) $(SERIAL_LIBS)
 usbdiskdirect_la_SOURCES = usbdiskdirect/linux.c
