diff -Naur -x '*~' libgphoto2-2.5.0/libgphoto2_port/libgphoto2_port/gphoto2-port-info-list.c libgphoto2-2.5.0-fix-format-errors/libgphoto2_port/libgphoto2_port/gphoto2-port-info-list.c
--- libgphoto2-2.5.0/libgphoto2_port/libgphoto2_port/gphoto2-port-info-list.c	2012-07-10 07:47:45.000000000 +0200
+++ libgphoto2-2.5.0-fix-format-errors/libgphoto2_port/libgphoto2_port/gphoto2-port-info-list.c	2012-08-04 20:47:53.480170660 +0200
@@ -306,7 +306,8 @@
 	if (result < 0)
 		return (result);
 	if (list->iolib_count == 0) {
-		gp_log (GP_LOG_ERROR, "No iolibs found in '%s'", iolibs);
+		gp_log (GP_LOG_ERROR, "gphoto2-port-info-list",
+                        "No iolibs found in '%s'", iolibs);
 		return GP_ERROR_LIBRARY;
 	}
         return (GP_OK);
